dask/order.py

```diff
@@ -315,6 +315,8 @@ def process_runnables() -> None:
                 while branches:
                     path = branches.popleft()
                     while True:
+                        # Loop invariant. Too expensive to compute at runtime
+                        # assert not set(known_runnable_paths).intersection(runnable_hull)
```

Comment:
this makes me want to introduce a `_validate` kwarg that allows us to enable these kinds of assertions without having them be active at runtime (alternatively via ctx variable)

Reply:
Yeah this makes sense I think, especially if we keep tinkering